mhkit/river/io/d3d.py

```diff
@@ -0,0 +1,408 @@
+        y_laydim=np.ma.getdata(data.variables[coords[1]][:], False)
+        points_laydim = np.array([ [x, y] for x, y in zip(x_laydim, y_laydim)])
+        
+        coords_request = str(data.variables[variable].coordinates).split()
+        x_wdim=np.ma.getdata(data.variables[coords_request[0]][:], False) 
+        y_wdim=np.ma.getdata(data.variables[coords_request[1]][:], False)
+        points_wdim=np.array([ [x, y] for x, y in zip(x_wdim, y_wdim)])
+        
+        bottom_depth_wdim = interp.griddata(points_laydim, bottom_depth, points_wdim)
+        
+        idx= np.where(np.isnan(bottom_depth_wdim))
+        
+        for i in idx: 
+            bottom_depth_wdim[i]= interp.griddata(points_laydim, bottom_depth,
+                                                  points_wdim[i], method='nearest')
+        
+        
+        
+    x_all=[]
+    y_all=[]
+    z_all=[]
+    v_all=[]
+    
+    N_layers = range(len(Layer_percentages))
+    for layer in N_layers:
+        layer_data= get_layer_data(data, variable, layer, time_index)
+        if layer_dim == 'wdim': 
+            z = [bottom_depth_wdim*Layer_percentages[layer]]
+        else: 
+            z = [bottom_depth*Layer_percentages[layer]]
+        x_all=np.append(x_all, layer_data.x)
+        y_all=np.append(y_all, layer_data.y)
+        z_all=np.append(z_all, z)
+        v_all=np.append(v_all, layer_data.v)
+    
+    known_points = np.array([ [x, y, z, v] for x, y, z, v in zip(x_all, y_all, z_all, v_all)])
+    
+    all_data= pd.DataFrame(known_points, columns=['x','y','z',f'{variable}'])
+    
+    return all_data
+
+
+def unorm(x, y ,z):
+    '''
+    Calculates the root mean squared value given three arrays. 
+
+    Parameters
+    ----------
+    x: array 
+        One input for the root mean squared calculation.(eq. x velocity) 
+    y: array
+        One input for the root mean squared calculation.(eq. y velocity) 
+    z: array
+        One input for the root mean squared calculation.(eq. z velocity) 
+
+    Returns
+    -------
+    unorm : array 
+        root mean squared output 
+    '''
+
+    assert isinstance(x,(np.ndarray, np.float64, pd.Series)), 'x must be an array'
+    assert isinstance(y,(np.ndarray, np.float64, pd.Series)), 'y must be an array'
+    assert isinstance(z,(np.ndarray, np.float64, pd.Series)), 'z must be an array'
+    
+    if len(x) == len(y) & len (y) ==len (z) :
+        xyz = np.array([x,y,z]) 
+        unorm = np.linalg.norm(xyz, axis= 0)
+    else:
+        raise Exception ('lengths of arrays do not mathch')
+    return unorm
+
+
+
+
+def turbulent_intensity(data, points='cells', time_index= -1, intermediate_values = False ):
+
+    '''
+    Calculated the turbulent intesity for a given data set for the specified points.  
+    Assumes variable names: turkin1, ucx, ucy and ucz.
+
+    Parameters
+    ----------
+    data : netcdf4 object 
+        After running a Delft3D model the output netcdf file can be read into this code. 
```

Comment:
This does not describe what data is. This statement is telling the user something that is possible which is not very useful if you are trying to figure out what data is. Please rewrite.